libcontainer/init_linux.go

```diff
@@ -88,6 +88,14 @@ func newContainerInit(t initType, pipe *os.File, consoleSocket *os.File, fifoFd,
 	}
 	switch t {
 	case initSetns:
+		// mountFiles is not needed in the setns case: close them all
```

Comment:
It looks like we do not ever send mount files in case of setns init, so why this code?

Reply:
I think it was added after this conversation: https://github.com/opencontainers/runc/pull/2576/files?file-filters%5B%5D=.c&file-filters%5B%5D=.go#r487807460

But I (and alban, talk to him just in case) agree that is not needed. I changed the code to return an error if the mountFiles slice has _some_ fd. It should never have one.

Due to how we construct the slice (with [make and a len][slice-make]) and that way is handy to [how we use it][slice-use] (it will be painful if we use struct literals to initialize it empty, because we will need to also [append in this case][append-continue] and it seems more error prone).

If we make sure the slice is empty when all pointers are nil, we can check here for `len(mountFiles) == 0`. But that makes the other code creating the slice painful to read (and as I mentioned, I think it makes sense to keep it like that). So, I preferred to keep that like that and here have a loop to check that all pointers in the slice are nil. 

Will push in a few minutes with the check, thanks! Also, let me know if you prefer to completely remove this code (I think detecting errors soon is nice, but your call :))

[slice-make]: https://github.com/opencontainers/runc/pull/2576/files?file-filters%5B%5D=.c&file-filters%5B%5D=.go#diff-08395501389b35b3ff26335eeaa81c2e113672e4feb4f500a3231e2f06128268R387
[slice-use]: https://github.com/opencontainers/runc/pull/2576/files?file-filters%5B%5D=.c&file-filters%5B%5D=.go#diff-08395501389b35b3ff26335eeaa81c2e113672e4feb4f500a3231e2f06128268R399
[append-continue]: https://github.com/opencontainers/runc/pull/2576/files?file-filters%5B%5D=.c&file-filters%5B%5D=.go#diff-08395501389b35b3ff26335eeaa81c2e113672e4feb4f500a3231e2f06128268R390